src/librustc_plugin/registry.rs

```diff
@@ -115,6 +118,17 @@ impl<'a> Registry<'a> {
         }));
     }
 
+    #[unstable(feature = "rustc_private", issue = "27812")]
+    #[rustc_deprecated(since = "1.15.0", reason = "replaced by macros 1.1 (RFC 1861)")]
+    pub fn register_custom_derive(&mut self, name: ast::Name, extension: SyntaxExtension) {
```

Comment:
Could you document here when an author should use this and what its effects are please?